app/api/admin_translations.py

```diff
@@ -0,0 +1,24 @@
+from flask import send_file, Blueprint, after_this_request
+import os
+import shutil
+
+
+app_dir = os.path.dirname(os.path.abspath(os.path.dirname(__file__)))
+file_dir = os.path.join(app_dir, 'translations')
+admin_blueprint = Blueprint('admin_blueprint', __name__, url_prefix='/admin/content/translations/all')
+
+
+@admin_blueprint.route('/', methods=['GET'])
+def download_translations():
+    shutil.make_archive('translations', 'zip', file_dir)
+    if os.path.exists(os.path.join(app_dir, 'translations.zip')):
+        os.remove(os.path.join(app_dir, 'translations.zip'))
+    shutil.move('translations.zip', app_dir)
+
+    @after_this_request
+    def remove_file(response):
+        os.remove(os.path.join(app_dir, 'translations.zip'))
+        return response
+    return send_file('translations.zip',  mimetype='application/zip',
```

Comment:
Move this line to correct place

Reply:
`@after_this_request` is getting skipped if I move the line upwards. You were referring to move the return statement above `@after_this_request`, right?  Did I get you wrong ?

Reply:
Yes I was referring to that

Reply:
Alright, will do it.

Reply:
Moved it to the right place .